### PR TITLE
[CHEC-667] - Bugfix/purge unused styles

### LINKED ISSUE #145: Tailwind is not purging unused styles because no template paths have been provided
image

Happens when running `yarn build-lib`

https://tailwindcss.com/docs/controlling-file-size/#removing-unused-css

### PR DESCRIPTION
**Pull request by jaepass**
- Added purge option to tailwind.config and included all template paths using custom tailwind classes

Closes #145

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -16,6 +16,17 @@ const fontSizes = {
...
 };
 module.exports = {
+  purge: {
+    enabled: process.env.NODE_ENV === 'production',
+    content: [
+      './components/**/*.vue',
+      './components/*.vue',
+      './components/*.js',
+      './stories/**/*.mdx',
+      './stories/**/**/*.mdx',
```
Comment (ScopeyNZ) [inline tailwind.config.js:26]: I don't think we should include these - as tailwind classes used here shouldn't be part of our bundle.

**Comment by ScopeyNZ** [comment]
Hmmmn. Unfortunately that breaks the deploy preview storybook. So maybe we need to set the purge paths differently when building the storybook? This has gotten a little challenging.

**Comment by jaepass** [comment]
hmmmmm is right, will continue to investigate!

**Comment by ScopeyNZ** [comment]
Okay I'm not sure how we can do this. We don't want to include stories in our configuration when building the bundle (`yarn build-lib`) but we do want to include it when building the storybook (`yarn storybook:build`). Maybe there's a way you can determine what command is running with an environment variable?

File: tailwind.config.js [hunk 2]
```diff
@@ -16,6 +16,16 @@ const fontSizes = {
   '5xl': '2.5rem',
 };
 module.exports = {
+  purge: {
+    enabled: process.env.NODE_ENV === 'production',
```
Comment (ScopeyNZ) [inline tailwind.config.js:20]: I _think_ Vue CLI takes care of enabling this for you in production already, so I don't think you need this line.

Comment (ScopeyNZ) [reply tailwind.config.js:20]: (Might be wrong)

Comment (jaepass) [reply tailwind.config.js:20]: yes you're right i did come across this, it does by default. after some digging, think the issue is that the tailwind extractor does not include mdx ext.

will try your above suggestion with seeing which command is running

**Comment by ScopeyNZ** [comment]
Just disabling the purgecss for storybook building is probably an acceptable workaround too.

**Comment by jaepass** [comment]
wait so looks like PurgeCss was added as a built-in in v1.4 which should then allow for the purge option in the configuration. https://github.com/tailwindcss/tailwindcss/releases
https://github.com/tailwindcss/tailwindcss/pull/1639
let me know if i should go ahead and upgrade to 1.4

**Comment by ScopeyNZ** [comment]
Oh right. That's probably worth doing. We've already updated dashboard a while back.

**Comment by jaepass** [comment]
oh shucks its still purging storybook...

File: tailwind.config.js [hunk 3]
```diff
@@ -15,7 +15,17 @@ const fontSizes = {
   '4xl': '2rem',
   '5xl': '2.5rem',
 };
+
 module.exports = {
+  purge: {
+    enabled: process.env.PURGE_ENABLED,
```
Comment (ScopeyNZ) [inline tailwind.config.js:21]:
```suggestion
    enabled: Boolean(process.env.PURGE_ENABLED),
```

**Comment by ScopeyNZ** [comment]
My bad - that should probably fix it.

**Comment by jaepass** [comment]
all good now thanks for you help on this!

**Comment by jaepass** [comment]
you're going to laugh to know i spent a good portion of last night adding those whitelist elements... lol

**Pull request merged**